fedimint-core/src/runtime.rs

```diff
@@ -0,0 +1,166 @@
+//! Copyright 2021 The Matrix.org Foundation C.I.C.//! Abstraction over an
+//! executor so we can spawn tasks under WASM the same way
+//! we do usually.
+
+// Adapted from https://github.com/matrix-org/matrix-rust-sdk
```

Comment:
How much adaptation did this require? Does it seem reasonable to work on extracting this into its own crate long term that can be used by Matrix, us, whoever else? Otherwise we should monitor changes to the original source in case they discover any bugs.

Reply:
https://docs.rs/prokio/latest/prokio/ is there, but doesn't look well maintained https://github.com/yewstack/prokio/issues